src/WebJobs.Extensions.DurableTask/ContextInterfaces/IDurableOrchestrationContext.cs

```diff
@@ -114,8 +114,9 @@ public interface IDurableOrchestrationContext
         /// <param name="method">HttpMethod used for api call.</param>
         /// <param name="uri">uri used to make the HTTP call.</param>
         /// <param name="content">Content passed in the HTTP request.</param>
+        /// <param name="retryOptions">The retry option for the HTTP task.</param>
         /// <returns>A <see cref="Task{DurableHttpResponse}"/>Result of the HTTP call.</returns>
-        Task<DurableHttpResponse> CallHttpAsync(HttpMethod method, Uri uri, string content = null);
+        Task<DurableHttpResponse> CallHttpAsync(HttpMethod method, Uri uri, string content = null, SerializableRetryOptions retryOptions = null);

```

Comment:
In the interest of reducing the breaking change impact and keeping our API surface fairly simple, I feel like I'd be fine leaving this simple `CallHttpAsync()` without the RetryOptions, and saying that RetryOptions is only available via `DurableHttpRequest.FailedRequestRetryOptions`.

If we do decide to include retryOptions here, we should rename it to `failedRequestRetryOptions` and add the C# docs from the DurableHttpRequest property to the param docs for this method.

Reply:
Fair enough; the reason I went with this approach is because the majority of the time I've used `CallHttpAsync` it's been through the `HttpMethod, Uri` overload. Any chance we have metrics/data (or can get some) on which of the overloads is used most often?

Reply:
Unfortunately, I don't believe we log the usage of CallHttpAsync(), and if we do, definitely not the distinction between these two method overloads, so I think we are going to have to go based on gut calls here.

In general, I have a preference towards keeping the simple overload simple and then having the `DurableHttpRequest` overload as the escape hatch for advanced functionality, but I'm open to the other option. @cgillum  and @bachuv, what are your thoughts?

Reply:
Based on [a GitHub code search](https://github.com/search?q=.Callhttpasync%28&type=Code) it definitely seems like the `HttpMethod, Uri` overload is the one users gravitate toward... still want to nix it?

Reply:
I’m not sure the best approach here. The advantage of adding this to the simple overload of that it’s more discoverable. If we think that it will be common for people configure retries, then making it more discoverable would be helpful. It’s also less code churn for existing codebases to adopt it. The disadvantage is that it’s technically a breaking change for anyone unit testing using their own implementation of this interface. That said, I’m leaning towards adding it to the simple overload because I feel like retries are a first class feature that should be configured broadly and isn’t niche.